lib/pos_blockchain.go

```diff
@@ -374,17 +366,51 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
 	return ancestors, nil
 }
 
-// addBlockToBlockIndex adds the block to the block index and uncommitted blocks map.
+// addBlockToBlockIndex adds the block to the block index with the given status. It also writes the block to the block
+// index in badger.
 func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock, blockStatus BlockStatus) error {
 	hash, err := desoBlock.Hash()
 	if err != nil {
 		return errors.Wrapf(err, "addBlockToBlockIndex: Problem hashing block %v", desoBlock)
 	}
 	// Need to get parent block node from block index
 	prevBlock := bc.blockIndex[*desoBlock.Header.PrevBlockHash]
-	bc.blockIndex[*hash] = NewBlockNode(prevBlock, hash, uint32(desoBlock.Header.Height), nil, nil, desoBlock.Header, blockStatus)
+	newBlockNode := NewBlockNode(prevBlock, hash, uint32(desoBlock.Header.Height), nil, nil, desoBlock.Header, blockStatus)
+	// Store the block in badger
+	err = bc.db.Update(func(txn *badger.Txn) error {
```

Comment:
I'm not a fan of how we currently handle these db operations in blockchain. Your code's fine; I just hate that there isn't a consolidated wrapper around this stuff. Or something that ensures we don't couple db operations directly with in-memory stuff.

Is this db write something that we may want to reuse in another place? Maybe we could refactor the db code in here to a single-purpose function that only updates the db. Like, `_dbStoreBlock`, or something like that. This way, it's slightly less coupled with the in-memory stuff, and makes the `addBlockToBlockIndex` a little clearer. I also think this function name doesn't imply you're going to do db operations. Maybe we call it like "storeBlock" or something, idk.